### DISCUSSION TITLE
Is it possible to integrate native code in Tauri?

### DISCUSSION THREAD

**Discussion by BraveEvidence** [rustdesk/rustdesk#543] [General]
I have never tried Tauri but this project looks better than electronjs which I haven't tried in a long time.

Is it possible to integrate C# code for windows and Swift for MacOS if a particular feature is not possible for Tauri like bluetooth?
I don't think electronjs support this of integrating native code.

I think the advantage of flutter or react native for desktop is they allow integrating native code.

**Comment by zuoxinyu** [comment]
It's possible to integrate native features in electron or tauri. For electron you could write native modules (via node-gyp or some same techology) And the same thing in tauri is [plugins](https://tauri.studio/v1/guides/features/plugin).

**Reply by BraveEvidence** [reply]
Thanks
